internal/controllers/machine/machine_controller.go

```diff
@@ -634,6 +634,12 @@ func (r *Reconciler) reconcileDelete(ctx context.Context, s *scope) (ctrl.Result
 }
 
 func (r *Reconciler) isNodeDrainAllowed(m *clusterv1.Machine) bool {
+	if util.IsControlPlaneMachine(m) && util.HasOwner(m.GetOwnerReferences(), clusterv1.GroupVersion.String(), []string{"KubeadmControlPlane"}) {
+		if _, exists := m.Annotations[clusterv1.PreTerminateDeleteHookAnnotationPrefix+"/kcp-cleanup"]; !exists {
```

Comment:
Q: we don't want to import controlplanev1 to prevent the go dependency right?

Reply:
Yes, that's correct.
We are trying to avoid having dependency from to core controller to KCP/CABPK